src/plugins/shape-generator/index.ts

```diff
@@ -0,0 +1,407 @@
+import { PluginController } from "#plugins/PluginController.js";
+import { ConfigItem } from "..";
+import { ellipseGeneratorExpressions, ellipseLatex } from "./latex/ellipse";
+import {
+  rectangleGeneratorExpressions,
+  rectangleLatex,
+} from "./latex/rectangle";
+import { DispatchedEvent } from "#globals";
+
+interface NewDropdownItem {
+  ariaLabel: string;
+  label: string;
+  okBtnExprId: string;
+  handler: () => void;
+  okBtnHandler: (expr: Desmos.ExpressionState) => void;
+}
+
+export default class ShapeGenerator extends PluginController<{
+  showSliders: boolean;
+}> {
+  static id = "shape-generator" as const;
+  static enabledByDefault = false;
+  static config: readonly ConfigItem[] = [
+    {
+      type: "boolean",
+      key: "showSliders",
+      default: false,
+    },
+  ];
+
+  private _isEditingShape = false;
+
+  private _onDispatchHandler = "";
+
+  private readonly newDropdownItems: NewDropdownItem[] = [
+    {
+      ariaLabel: "Add ellipse",
+      label: "ellipse",
+      okBtnExprId: "shape-generator-ellipse",
+      handler: () => {
+        this.calc.setExpressions(ellipseGeneratorExpressions);
+      },
+      okBtnHandler: (expr) => {
+        const xHelper = this.calc.HelperExpression({
+          latex: "x_{ellipseGenerator}",
+        });
+        const yHelper = this.calc.HelperExpression({
+          latex: "y_{ellipseGenerator}",
+        });
+        const rxHelper = this.calc.HelperExpression({
+          latex: "r_{xEllipseGenerator}",
+        });
+        const ryHelper = this.calc.HelperExpression({
+          latex: "r_{yEllipseGenerator}",
+        });
+        const angleHelper = this.calc.HelperExpression({
+          latex: "A_{ellipseGenerator}",
+        });
+
+        angleHelper.observe("numericValue", () => {
+          const x = xHelper.numericValue;
+          const y = yHelper.numericValue;
+          const rx = rxHelper.numericValue;
+          const ry = ryHelper.numericValue;
+          const angle = angleHelper.numericValue;
+
+          if (isNaN(x) || isNaN(y) || isNaN(rx) || isNaN(ry) || isNaN(angle)) {
+            return;
+          }
+
+          // TODO: HelperExpression is missing unobserve types
```

Comment:
pending TODO (another one below)

Reply:
Oh, but that's an issue of `HelperExpression`s missing the type annotations for the `unobserve` function. I'm not sure what the best way to type this would be, to simply copy-paste the `unobserve` types from `Calc`? Or to make a shared type like `interface Observable` that the rest of types extend?

Reply:
Oh, that's tricky since the types are in the `desmos` package (which is 4 years old, incorrect in parts, and NOT published by Desmos). I'd probably just toss in an `as any` and a comment explaining that the `desmos` package is wrong here.

Reply:
Alright, I guess I'll type assert for now. But maybe a PR could be made to `@types/desmos` fixing these types? Or just fork the package, and add it to the DesModder monorepo, and fix them there.